purchase_partial_invoicing/purchase.py

```diff
@@ -33,6 +33,25 @@ def _invoiced_qty(self, cursor, user, ids, name, arg, context=None):
             res[line.id] = invoiced_qty
         return res
 
+    def _fully_invoiced(self, cursor, user, ids, name, arg, context=None):
+        res = {}
+        for line in self.browse(cursor, user, ids, context=context):
+            res[line.id] = False
+            if line.invoiced_qty == line.product_qty:
+                res[line.id] = True
+        return res
+
+    def _all_invoices_approved(self, cursor, user, ids, name, arg, context=None):
+        res = {}
+        for line in self.browse(cursor, user, ids, context=context):
+            res[line.id] = False
+            if line.fully_invoiced:
+                for invoice_line in line.invoice_lines:
+                    if invoice_line.invoice_id.state not in ['draft',
+                                                             'cancel']:
+                        res[line.id] = True
```

Comment:
As @pedrobaeza said, you should inverse the logic and break if any invoice is in `draft` or `cancel` state